planner/core/planbuilder.go

```diff
@@ -465,7 +471,7 @@ func isPrimaryIndex(indexName model.CIStr) bool {
 	return indexName.L == "primary"
 }
 
-func getPossibleAccessPaths(indexHints []*ast.IndexHint, tblInfo *model.TableInfo) ([]*accessPath, error) {
+func getPossibleAccessPaths(indexHints []*ast.IndexHint, hintFromComment []*ast.IndexHint, tblInfo *model.TableInfo) ([]*accessPath, error, []error) {
```

Comment:
It makes me confused to return both `error` and `[]error`. Can we append warnings in this function? We can make `getPossibleAccessPaths` to `(b *PlanBuilder) getPossibleAccessPaths` to get the `context`.

Reply:
okay, it's a good solution